tests/phpmailer_test.php

```diff
@@ -0,0 +1,68 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * auth_saml2 PHPMailer unit tests
+ *
+ * @package    auth_saml2
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * auth_saml2 PHPMailer unit tests
+ *
+ * @package    auth_saml2
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class auth_saml2_phpmailer_testcase extends advanced_testcase {
```

Comment:
In out particular case we are trying to check PHPMailer, however, I'm thinking if potentially we can add more checks that can be run after we simplesamlphp upgrade library without creating a test file for each of the case. 

Do you think it makes sense to rename this test file to something a bit more generic? E.g. auth_saml2_simplesamlphp_upgrade_testcase or something...

Reply:
Thank you @dmitriim for suggesting that !! Renamed file and class name. Please review. Thank you..